e2e/drift/drift_test.go

```diff
@@ -32,13 +33,24 @@ var _ = Describe("Drift", Ordered, func() {
 			b := getBundle(bundleName, k)
 			return b.Status.Summary.Ready == 1
 		}).Should(BeTrue())
+
+		defer func() {
+			if r := recover(); r != nil {
+				bundle := getBundle(bundleName, k)
+				GinkgoWriter.Printf("bundle status: %v", bundle.Status)
+			}
+		}()
 	})
 
 	AfterEach(func() {
-		out, err := k.Namespace(env.Namespace).Delete("-f", testenv.AssetPath(asset))
+		out, err := k.Namespace(env.Namespace).Delete("-f", testenv.AssetPath(asset), "--wait")
```

Comment:
We are waiting on some namespace deletions, but not on others? I guess this corresponds to random/static name, or is it by usage?

Reply:
In this case, we wait for complete deletion of a `gitrepo`, to be absolutely sure that no deletion will still be in progress while subsequent test cases run, to limit possible conflicts/interferences between test cases. The idea is to prevent failures like [this one](https://github.com/rancher/fleet/actions/runs/9655199796/job/26630714091), with bundles sharding names between test cases.

Reply:
Ah, we can't use a random namespace, because the cluster must be in the same namespace.